Sources/UI/ScreenViewController.swift

```diff
@@ -91,11 +94,14 @@ open class ScreenViewController: UICollectionViewController, UICollectionViewDat
     override open func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
         
-        let userInfo: [String: Any] = [
+        let userInfoWithOptionalCampaignId: [String: Any?] = [
```

Comment:
This approach is a bit odd to me. Why not make it of type [String: Any] and optionally add the `campaignID` key if it exists. That way you don't have to call `compactMapValues { $0 }` in multiple places.

Reply:
👍